source/core/datasource/handler/TimeSeriesDataSourceHandler.js

```diff
@@ -22,12 +22,13 @@ class TimeSeriesDataSourceHandler extends DataSourceHandler {
         }, connector);
 
         const lastStartTimeCst = this.parser && this.lastStartTime || properties.startTime;
+        const endpoint = properties.protocol + '://' + properties.endpointUrl;
+
         this.connector.onReconnect = () => {
             // if not real time, preserve last timestamp to reconnect at the last time received
             // for that, we update the URL with the new last time received
             if (lastStartTimeCst !== 'now') {
-                this.connector.setUrl(this.parser.buildUrl(
-                    {
+                this.connector.setUrl(endpoint + '?' + this.parser.buildUrl({
```

Comment:
parser object of this is sometimes undefined, therefore invocation of buildUrl causes exception: 

`TimeSeriesDataSourceHandler.js:35 
        
       Uncaught TypeError: Cannot read properties of undefined (reading 'buildUrl')
    at connector.onReconnect (TimeSeriesDataSourceHandler.js:35:68)
    at WebSocketConnector.createReconnection (WebSocketConnector.js:132:64)
    at ws.onclose (WebSocketConnector.js:108:26)`